### PR TITLE
Add kubectl rollout

### PR DESCRIPTION
**Pull request by janetkuo**
`kubectl rollout` contains related sub-commands, such as `kubectl rollout history`, `kubectl rollout undo`, `kubectl rollout pause`, `kubectl rollout resume`.

Addresses #17168; separated from #19835 for the development of other rollout sub-commands. 
See [proposal](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/deploy.md#deployment-version). 

cc @bgrant0607 @nikhiljindal @ironcladlou @kargakis @kubernetes/sig-config @kubernetes/kubectl @madhusudancs

### CODE REVIEW

File: pkg/kubectl/cmd/rollout/rollout.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+const (
+	rollout_long            = `rollout manages a deployment using subcommands`
```
Comment (kargakis) [inline pkg/kubectl/cmd/rollout/rollout.go]: rollout will not work only for deployments from my understanding.

Comment (kargakis) [inline pkg/kubectl/cmd/rollout/rollout.go]: But I guess it's fine for now

**Comment by kargakis** [comment]
LGTM

Let's get this in so we can work independently on the subcommands

File: pkg/kubectl/cmd/rollout/rollout.go (hunk 1 above)
Comment (janetkuo) [inline pkg/kubectl/cmd/rollout/rollout.go]: Yes we can update this when the support for other resources is in.

**Comment by janetkuo** [comment]
Applying LGTM from @kargakis's comments.

**Comment by bgrant0607** [comment]
@k8s-bot test this

**Comment by janetkuo** [comment]
@k8s-bot e2e test this

**Pull request merged**